BiometricLoginKotlin/app/src/main/java/com/example/biometricloginsample/LoginActivity.kt

```diff
@@ -55,33 +70,15 @@ class LoginActivity : AppCompatActivity() {
      */
     override fun onResume() {
         super.onResume()
-        val ciphertextWrapper = cryptographyManager.getCiphertextWrapperFromSharedPrefs(
-            applicationContext,
-            SHARED_PREFS_FILENAME,
-            Context.MODE_PRIVATE,
-            CIPHERTEXT_WRAPPER
-        )
 
         if (ciphertextWrapper != null) {
             if (SampleAppUser.fakeToken == null) {
-                encryptedServerTokenWrapper = ciphertextWrapper
                 showBiometricPromptForDecryption()
             } else {
                 // The user has already logged in, so proceed to the rest of the app
                 // this is a todo for you, the developer
                 updateApp(getString(R.string.already_signedin))
             }
-        } else {
-            setupForLoginWithPassword()
-        }
-
-        binding.useBiometrics.setOnClickListener {
-            if (ciphertextWrapper != null) {
-                encryptedServerTokenWrapper = ciphertextWrapper
-                showBiometricPromptForDecryption()
-            } else {
-                startActivity(Intent(this, EnableBiometricLoginActivity::class.java))
-            }
         }
```

Comment:
I wonder if this logic can't be moved into the ViewModel (the business logic part) and have the Activity only react to state changes.